### ISSUE TITLE
New rules: selector-id-pattern and selector-class-pattern

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#431]
Specify pattern of selectors

Options:

```js
string
```

regex (eg: `/\.([a-z]+-)?[A-Z][a-z]+(-[a-z][a-zA-Z]+)(--[a-z][a-zA-Z]+)/` to match `.org-Block-eleMent--modifier`.

**Comment by davidtheclark** [comment]
The key decision here is: what segment of a selector does the pattern apply to? I'm thinking compound selectors (checks chained simple selectors but does not span combinators). Thoughts?

**Comment by jeddy3** [comment]
Sounds good. Lets not include _complex selectors_.

Do we need to update the name to reflect this? e.g.

```
selector-simple-and-compound-pattern
```

?

**Comment by davidtheclark** [comment]
Sure. Verbose but accurate.

**Comment by davidtheclark** [comment]
I've changed my mind. Right now I'm thinking just `selector-simple-pattern` would be better. That would provide for the simplest configuration. Any disagreement?

**Comment by jeddy3** [comment]
> Any disagreement?

Sgtm.

**Comment by davidtheclark** [comment]
I've decided to break this down further, into `selector-id-pattern` and `selector-class-pattern`. I figure that there's actually no need for anybody to write a pattern for every other simple selector (universal, type, pseudo class), except *maybe* attribute --- in which case we could add another rule. But after coding a bit I realized this was the simplest route.

**Comment by jeddy3** [comment]
> in which case we could add another rule. But after coding a bit I realized this was the simplest route.

Makes sense!
